tests/marqo_test.py

```diff
@@ -180,24 +199,31 @@ def warm_request(self, func, *args, **kwargs):
             func(*args, **kwargs)
 
     def create_cloud_index(self, index_name, settings_dict=None, **kwargs):
+        """
+        Creates a cloud index with the given name and settings. If settings_dict or any kwargs are provided,
+        they will be used to create a unique index name by hashing the settings_dict and kwargs.
+        If settings_dict is not provided, the index name will be the given index_name + the index_suffix.
+
+        Performs check for status of index and waits for index to be ready before returning.
+        """
         client = marqo.Client(**self.client_settings)
-        settings_dict = settings_dict if settings_dict else {}
         index_name = f"{index_name}-{self.index_suffix}"
         if settings_dict or kwargs:
             index_name = f"{index_name}-{create_settings_hash(settings_dict, kwargs)}"
-        settings_dict.update({
-            "inference_type": "marqo.CPU", "storage_class": "marqo.basic", "model": "hf/all_datasets_v4_MiniLM-L6"
-        })
+        if settings_dict:
+            settings_dict.update({
+                "inference_type": "marqo.CPU", "storage_class": "marqo.basic"
+            })
         try:
-            status = client.http.get(f"/indexes/{index_name}/status")["index_status"]
+            status = client.http.get(f"indexes/{index_name}/status")["index_status"]
             if status == "CREATING":
-                while status == "CREATING":
-                    time.sleep(10)
-                    status = client.http.get(f"/indexes/{index_name}/status")["index_status"]
+                cloud_wait_for_index_status(client.http, index_name, "READY")
             if status != "READY":
```

Comment:
shouldn't this be `else if`? if status == "CREATING" and then we wait for the index to be created, then the second if statement runs and then the index is attempted to be created again - which probably raises a MarqoWebError, which gets caught in the except block and then the index is attempted to be created a 3rd time.

It is worth re-reviewing this block of code to iron out any odd logic

Reply:
in the first part we are checking if index status is CREATING and if so we are entering infinite while loop until index reaches READY, in the end it will just ignore next check if it is READY, else it will trigger creation for that index, it was working fine this way

Reply:
Yep all good I think I misread your code